src/mgr/MgrClient.h

```diff
@@ -116,13 +118,18 @@ class MgrClient : public Dispatcher
   bool handle_mgr_close(MMgrClose *m);
   bool handle_command_reply(MCommandReply *m);
 
-  void send_perf_metric_query();
-  void set_perf_metric_query_cb(std::function<MMgrReport*()> cb_)
+  void set_set_perf_queries_cb(std::function<void(
+                                 const std::list<OSDPerfMetricQuery> &)> cb)
   {
       Mutex::Locker l(lock);
-      perf_metric_query_cb = cb_;
+      set_perf_queries_cb = cb;
   }
 
+  void set_get_perf_report_cb(std::function<void(OSDPerfMetricReport *)> cb)
```

Comment:
Nit: I would just combine them into a single method w/ two parameters since they are really tied together.

Reply:
@colletj would you want to address this in your #24180 and then I just would rebase, or I can do it here myself.

Reply:
@trociny no problem, i'll take care of this.